### PR TITLE
FOUR-7965 add device preview behavior

### PR DESCRIPTION
**Pull request by devmiguelangel**
## Issue & Reproduction Steps

Expected behavior: 
Preview the screen in desktop and mobile resolutions

Actual behavior: 
N/A

## Solution
Add a desktop and mobile preview button next to the PREVIEW button

## How to Test
1. Open a Screen builder
2. Press on mobile preview button next to the preview button
3. The mobile preview must be displayed

## Related Tickets & Packages
[FOUR-7965](https://processmaker.atlassian.net/browse/FOUR-7965)

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

<sup> Created from VS Code using [CodeStream](https://codestream.com/?utm_source=cs&utm_medium=pr&utm_campaign=github*com)</sup>

[FOUR-7965]: https://processmaker.atlassian.net/browse/FOUR-7965?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/deviceConstants.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export const MAX_MOBILE_WIDTH = 480;
```
Comment (josechirivella) [inline src/deviceConstants.js:1]: Just add this to the Component at the top. A file just for one constant seems like a bit much

File: src/components/vue-form-renderer.vue [hunk 2]
```diff
@@ -88,6 +99,24 @@ export default {
       scrollable: null,
     };
   },
+  computed: {
+    containerClass() {
+      return this.parentScreen ? `screen-${this.parentScreen}` : 'custom-css-scope';
+    },
+    cssDevice() {
+      const { devicePixelRatio } = window;
+      const deviceDensity = devicePixelRatio === undefined ? 1 : devicePixelRatio;
```
Comment (josechirivella) [inline src/components/vue-form-renderer.vue:108]: do we care about the device pixel ratio? Seems like we just care about the device width?

File: .prettierrc [hunk 3]
```diff
@@ -1,3 +1,5 @@
 {
-  "trailingComma": "none"
+  "trailingComma": "all",
+  "singleQuote": true,
+  "printWidth": 120
```
Comment (josechirivella) [inline .prettierrc:4]: make this 140, 120 is too short

File: .eslintrc.js [hunk 4]
```diff
@@ -20,13 +20,12 @@ module.exports = {
...
 
   rules: {
-    "prettier/prettier": ["error", { trailingComma: "none" }],
     "no-unexpected-multiline": "error",
     "no-param-reassign": 1,
     eqeqeq: "error",
-    "max-len": ["error", { code: 140, ignoreUrls: true }],
-    "comma-dangle": ["error", "never"],
-    quotes: ["error", "double", { allowTemplateLiterals: true }],
+    "max-len": ["error", { code: 120, ignoreUrls: true }],
```
Comment (josechirivella) [inline .eslintrc.js:26]: 140 here also

File: src/deviceConstants.js (hunk 1 above)
Comment (devmiguelangel) [reply src/deviceConstants.js:1]: It will be removed

File: src/components/vue-form-renderer.vue (hunk 2 above)
Comment (devmiguelangel) [reply src/components/vue-form-renderer.vue:108]: You are right. It will be removed

File: .eslintrc.js (hunk 4 above)
Comment (devmiguelangel) [reply .eslintrc.js:26]: It will be fixed

**Review by josechirivella — APPROVED** [review]

**Review by danloa — COMMENTED** [review]
@devmiguelangel  the changes of App.vue must be present in PM Core too (file resources/js/processes/screen-builder/screen.vue).

**Comment by devmiguelangel** [comment]
> @devmiguelangel the changes of App.vue must be present in PM Core too (file resources/js/processes/screen-builder/screen.vue).

@danloa The changes wore applied in the PM Core #1389

**Review by danloa — APPROVED** [review]

**Pull request merged**